src/contexts/AuthContext.tsx

```diff
@@ -40,22 +40,20 @@ const AuthProvider: React.FC<Props> = ({children}: Props) => {
 
         // Check if user is an admin user
         firestore
-          .collection('admin')
-          .doc('users')
+          .collection('users')
+          .doc(user.uid)
           .get()
           .then(doc => {
             if (doc.exists) {
-              // Get the document data that contains all admin userIds
               const userData = doc.data();
               if (userData) {
-                const adminUserIds: string[] = userData.userId ?? [];
-                if (adminUserIds.includes(user.uid)) setIsAdmin(true);
+                if (userData.admin) setIsAdmin(true);
```

Comment:
Does caching isAdmin locally allow someone malicious to change their status? It probably doesn't cause we can do serverside checks too on any database calls but just wondering?

Reply:
My understanding is that a user cannot modify internal React variables but also yes, the server-side Firestore rules restrict reading and writing Firestore docs based on what Firestore says about admin status